WordPress/Classes/ViewRelated/Stats/Shared Views/Stats Detail/SiteStatsInsightsDetailsViewModel.swift

```diff
@@ -367,7 +367,17 @@ class SiteStatsInsightsDetailsViewModel: Observable {
                 return rows
             }
         case .insightsCommentsAuthors, .insightsCommentsPosts, .insightsCommentsTotals:
-            return insightsImmuTable(for: (.allComments, insightsStore.allCommentsInsightStatus)) {
+            /// Comments depend both on PeriodStore and InsightsStore states
+            let status: StoreFetchingStatus = {
+                if insightsStore.allCommentsInsightStatus == .loading {
```

Comment:
Could this be removed since the `else` condition will also return `.loading` if `insightsStore.allCommentsInsightStatus == .loading`? Or is the current logic needed to ensure this check takes precedence over the `periodStore.getSummary() != nil` check below?

Reply:
Yes, you're right, it's because of the precedence, so the loading state would be represented while the other data for total comments details is loading, since `periodStore.getSummary()` should always have value when we come from the main screen to details.

Reply:
Thanks for explaining 👍